io/FV3GFS_restart_io.F90

```diff
@@ -0,0 +1,920 @@
+      nt=0
+      nt=nt+1 ; sfc_name2(nt) = 'slmsk'
+      nt=nt+1 ; sfc_name2(nt) = 'tsea'    !tsfc
+      nt=nt+1 ; sfc_name2(nt) = 'sheleg'  !weasd
+      nt=nt+1 ; sfc_name2(nt) = 'tg3'
+      nt=nt+1 ; sfc_name2(nt) = 'zorl'
+      nt=nt+1 ; sfc_name2(nt) = 'alvsf'
+      nt=nt+1 ; sfc_name2(nt) = 'alvwf'
+      nt=nt+1 ; sfc_name2(nt) = 'alnsf'
+      nt=nt+1 ; sfc_name2(nt) = 'alnwf'
+      nt=nt+1 ; sfc_name2(nt) = 'facsf'
+      nt=nt+1 ; sfc_name2(nt) = 'facwf'
+      nt=nt+1 ; sfc_name2(nt) = 'vfrac'
+      nt=nt+1 ; sfc_name2(nt) = 'canopy'
+      nt=nt+1 ; sfc_name2(nt) = 'f10m'
+      nt=nt+1 ; sfc_name2(nt) = 't2m'
+      nt=nt+1 ; sfc_name2(nt) = 'q2m'
+      nt=nt+1 ; sfc_name2(nt) = 'vtype'
+      nt=nt+1 ; sfc_name2(nt) = 'stype'
+      nt=nt+1 ; sfc_name2(nt) = 'uustar'
+      nt=nt+1 ; sfc_name2(nt) = 'ffmm'
+      nt=nt+1 ; sfc_name2(nt) = 'ffhh'
+      nt=nt+1 ; sfc_name2(nt) = 'hice'
+      nt=nt+1 ; sfc_name2(nt) = 'fice'
+      nt=nt+1 ; sfc_name2(nt) = 'tisfc'
+      nt=nt+1 ; sfc_name2(nt) = 'tprcp'
+      nt=nt+1 ; sfc_name2(nt) = 'srflag'
+      nt=nt+1 ; sfc_name2(nt) = 'snwdph'  !snowd
+      nt=nt+1 ; sfc_name2(nt) = 'shdmin'
+      nt=nt+1 ; sfc_name2(nt) = 'shdmax'
+      nt=nt+1 ; sfc_name2(nt) = 'slope'
+      nt=nt+1 ; sfc_name2(nt) = 'snoalb'
+      !--- variables below here are optional
+      nt=nt+1 ; sfc_name2(nt) = 'sncovr'
+      nt=nt+1 ; sfc_name2(nt) = 'snodl' !snowd on land portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'weasdl'!weasd on land portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'tsfc'  !tsfc composite
+      nt=nt+1 ; sfc_name2(nt) = 'tsfcl' !temp on land portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'zorlw' !zorl on water portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'zorll' !zorl on land portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'zorli' !zorl on ice portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'albdirvis_lnd'
+      nt=nt+1 ; sfc_name2(nt) = 'albdirnir_lnd'
+      nt=nt+1 ; sfc_name2(nt) = 'albdifvis_lnd'
+      nt=nt+1 ; sfc_name2(nt) = 'albdifnir_lnd'
+      nt=nt+1 ; sfc_name2(nt) = 'emis_lnd'
+      nt=nt+1 ; sfc_name2(nt) = 'emis_ice'
+      nt=nt+1 ; sfc_name2(nt) = 'sncovr_ice'
+      nt=nt+1 ; sfc_name2(nt) = 'snodi' ! snowd on ice portion of a cell
+      nt=nt+1 ; sfc_name2(nt) = 'weasdi'! weasd on ice portion of a cell
+
+      if (Model%use_cice_alb .or. Model%lsm == Model%lsm_ruc) then
+        nt=nt+1 ; sfc_name2(nt) = 'albdirvis_ice'
+        nt=nt+1 ; sfc_name2(nt) = 'albdifvis_ice'
+        nt=nt+1 ; sfc_name2(nt) = 'albdirnir_ice'
+        nt=nt+1 ; sfc_name2(nt) = 'albdifnir_ice'
+!        nt=nt+1 ; sfc_name2(nt) = 'sfalb_ice'
+      endif
+
+      if(Model%cplwav) then
+        sfc_name2(nvar_s2m) = 'zorlwav' !zorl from wave component
+      endif
+
+      nt = nvar_s2m ! next variable will be at nvar_s2m
+
+      !--- NSSTM inputs only needed when (nstf_name(1) > 0) .and. (nstf_name(2)) == 0)
+      nt=nt+1 ; sfc_name2(nt) = 'tref'
```

Comment:
In the future we may want to use an array of variable names and run the nt=nt+1 ; sfc_Name2(nt) = '...' in a loop sized to the length of that array of variables. Especially with the other sets like this too.